### PR TITLE
XSS protection + added the ability to create posts without tags

### PR DESCRIPTION
**Pull request by tavriaforever**
// cc @tadatuta @tormozz48 @gela-d

### CODE REVIEW

File: desktop.blocks/forum/forum.bemhtml [hunk 1]
```diff
@@ -1,3 +1,10 @@
 block('forum')(
-    js()(true)
+    // for check, if user checked at least one label
+    // for create/edit issue forms - knowledge is taken
+    // from common config website
+    js()(function() {
+        return {
+            labelsRequired: this.ctx.forum.labelsRequired
+        }
```
Comment (tadatuta) [inline desktop.blocks/forum/forum.bemhtml]: ;

File: desktop.blocks/issue/issue.bemtree [hunk 2]
```diff
@@ -360,7 +358,8 @@ block('issue')(
     // body content
     elem('body')(
         content()(function() {
-            var text = this._issue.body;
+            var text = _.escape(this._issue.body); // escape for protection from xss
```
Comment (tadatuta) [inline desktop.blocks/issue/issue.bemtree]: should be `escape to protect` but it's kinda obvious

File: README.ru.md [hunk 3]
```diff
@@ -29,14 +29,37 @@ bem-forum
...
 ```
-"forum": {
-    "storage": {
-        "user": "bem",
-        "repo": "bem-talk"
+{
+    "forum": {
+        "auth": {
+            "tokens": [
+                /* Для получения данных с гитхаба для неавторизованных пользователей,
```
Comment (tadatuta) [inline README.ru.md]: extra comma

File: README.ru.md [hunk 4]
```diff
@@ -29,14 +29,37 @@ bem-forum
...
+                 * используются сгенерированные токены, которые увеличивают лимит обращений к API
+                 * c 60 запросов в час до 5000. В случае высокой активности на форуме,
+                 * нужно добавить дополнительные токены.
+                 */
+                "7fdffdd7a38fd8a205fdf5fc910f35d3bfd05341" // данный токен только для демонстрации
+            ]
+        },
+        /* Чтобы добавлять метки(labels), нужно добавить токен, который имеет права на git push
```
Comment (tadatuta) [inline README.ru.md]: space before parenthesis

File: README.ru.md [hunk 5]
```diff
@@ -75,8 +98,14 @@ bem-forum
 }
 ```
 
-### Использование форума как плагина к существующему приложению
+### p.s. Важно о метках
+
+Стандартное поведение из "коробки" – указать метку(label) при создании/редактировании поста *нельзя*
```
Comment (tadatuta) [inline README.ru.md]: "" -> «»

File: desktop.blocks/forum/forum.js [hunk 6]
```diff
@@ -74,7 +74,11 @@ modules.define('forum', ['i-bem__dom', 'jquery'], function(provide, BEMDOM, $) {
         },
 
         _toggleFormAdd: function() {
-            this.findBlockInside('labels', 'forum-labels').getLabels();
+            var labels = this.findBlockInside('labels', 'forum-labels');
+
+            if(labels) {
+                labels.getLabels()
```
Comment (tadatuta) [inline desktop.blocks/forum/forum.js]: ;

File: configs/common/node.json [hunk 7]
```diff
@@ -2,15 +2,14 @@
     "forum": {
         "auth": {
             "tokens": [
-                "7a2fb4d7a380f8a20562f5fc910f35d3b1605341",
-                "db2b748a0bcef3ef2f009bdcc425c86c9ab37770",
-                "265a82cf53438f87c98f0cbf0ba8aec68d14d761"
+                "7a2fb4d7a380f8a20562f5fc910f35d3b1605341"
```
Comment (gela-d) [inline configs/common/node.json:5]: Do we need array here?

P.S. If we don't, we shoud correct README.md

File: desktop.blocks/forum/forum.bemhtml (hunk 1 above)
Comment (tavriaforever) [inline desktop.blocks/forum/forum.bemhtml]: ok

File: desktop.blocks/issue/issue.bemtree (hunk 2 above)
Comment (tavriaforever) [inline desktop.blocks/issue/issue.bemtree]: ok, I will remove comments

File: README.ru.md (hunk 3 above)
Comment (tavriaforever) [inline README.ru.md]: ok

File: README.ru.md (hunk 4 above)
Comment (tavriaforever) [inline README.ru.md]: ?

File: README.ru.md (hunk 5 above)
Comment (tavriaforever) [inline README.ru.md]: ok

File: desktop.blocks/forum/forum.js (hunk 6 above)
Comment (tavriaforever) [inline desktop.blocks/forum/forum.js]: ok

File: configs/common/node.json (hunk 7 above)
Comment (tavriaforever) [inline configs/common/node.json:5]: We need array here in any cases

**Comment by tavriaforever** [comment]
up

File: README.ru.md (hunk 4 above)
Comment (tadatuta) [inline README.ru.md]: метки(labels) -> метки (labels)

**Comment by tavriaforever** [comment]
up2

**Pull request merged**